src/Octoshift/VersionChecker.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Reflection;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace OctoshiftCLI
+{
+    public class VersionChecker
+    {
+        private string _latestVersion;
+
+        public async Task<bool> IsLatest()
+        {
+            var curVersion = GetCurrentVersion();
+            var latestVersion = await GetLatestVersion();
+
+            curVersion = curVersion[..latestVersion.Length];
+
+            return curVersion == latestVersion;
+        }
+
+        public string GetCurrentVersion()
+        {
+            return Assembly.GetExecutingAssembly().GetName().Version.ToString();
+        }
+
+        public async Task<string> GetLatestVersion()
+        {
+            if (string.IsNullOrWhiteSpace(_latestVersion))
+            {
+                using var client = new HttpClient();
+                client.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
+                client.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("OctoshiftCLI", GetCurrentVersion()));
```

Comment:
This instantiation should go into the `ctor`.  I would also inject it into the `ctor` just like the way we do in out Api classes.

Reply:
I just made this change and I don't like it. I don't like concepts like HTTP headers leaking out, IMO any knowledge about headers and how they are constructed should be encapsulated in this class (just like how we do with the auth header).

Reply:
That's fine, but I'd say def. get rid of that dependency, it's really unnecessary to be dependent on it just to get the assembly version.

Reply:
Also regarding the extension, I see it just as a helper nothing more, the framework itself has lots of extensions like that. But again I am fine either way 😃

Reply:
my opinion on this depends on what happens with the other comment thread above about the VersionChecker dependency

Reply:
injecting IVersionProvider instead now